netanalysis/ooni/data/sync_measurements.py

```diff
@@ -59,6 +59,13 @@ class CostLimitError(Exception):
     def __init__(self, message: str) -> None:
         super().__init__(message)
 
+def _make_local_path(output_dir: pathlib.Path, entry: ooni_client.FileEntry) -> pathlib.Path:
+    basename = pathlib.PurePosixPath(entry.url.path).name
+    # Convert .json.lz4 and .tar.lz4 filenames.
+    if not basename.endswith('.jsonl.gz'):
+        basename = basename.rsplit('.', 2)[0] + '.jsonl.gz'
+    return output_dir / entry.country / entry.test_type / f'{entry.date:%Y-%m-%d}' / basename
```

Comment:
Is this coercing the `/` to a string when concatenating?

I found it confusing to read. I would just make the whole return into a single format string.

Reply:
`output_dir` is of type Path. That's how you join path components to return a Path, not a string.

Reply:
Also, formatted strings is not portable for paths. This will work on both Windows and Linux.